src/database/queries/baseFieldLocalizations/createOrUpdateByPrimaryKey.sql

```diff
@@ -0,0 +1,16 @@
+INSERT INTO base_field_localizations (
+  base_field_id,
+  language,
+  label,
+  description
+) VALUES (
+  :baseFieldId,
+  :language,
+  :label,
+  :description
+)
+ON CONFLICT (base_field_id, language)
+DO UPDATE SET
```

Comment:
Neat, this only affects the specific row with that matching key information, right?

I admit it's very nerve racking to me to se an `UPDATE SET {...}` without a `WHERE` <-- since without the rows being specified that would update EVERY ROW / blow away a ton of data.  I almost wanna suggest adding the WHERE to be redundantly defensive...

@jasonaowen do you have any thoughts?  This is our first upsert so lets make sure the pattern is what we want!

Reply:
@hminsky2002 would you be open to adding a test to make sure that PUT against an existing localization *only* updates the specified localization?  We should probably have that test regardless.

Reply:
I just double checked the [PostgreSQL docs for the `ON CONFLICT` clause](https://www.postgresql.org/docs/current/sql-insert.html#SQL-ON-CONFLICT), and yes, only the specific matching row is updated:

> `ON CONFLICT DO UPDATE` updates the existing row that conflicts with the row proposed for insertion as its alternative action.

There is a `WHERE` condition allowed, which further restricts whether or not an existing row already found to be conflicting should be updated. Their example:

```SQL
-- Don't update existing distributors based in a certain ZIP code
INSERT INTO distributors AS d (did, dname) VALUES (8, 'Anvil Distribution')
    ON CONFLICT (did) DO UPDATE
    SET dname = EXCLUDED.dname || ' (formerly ' || d.dname || ')'
    WHERE d.zipcode <> '21201';
```

So yes, I think this is the correct way to write this! Adding a test to make that clearer is probably a good idea, as you suggest.